src/main/scala/com/evolutiongaming/concurrent/FutureHelper.scala

```diff
@@ -47,10 +47,8 @@ object FutureHelper {
     }
 
     def sequenceSuccessful[A, M[X] <: TraversableOnce[X]](in: M[Future[A]])(implicit cbf: CanBuildFrom[M[Future[A]], A, M[A]], executor: ExecutionContext): Future[M[A]] = {
-      def wrap(f: Future[A]): Future[Option[A]] = f.map(Some.apply).recover { case NonFatal(_) => None }
-
       in.foldLeft(Future.successful(cbf(in))) {
-        (acc, f) => for (r <- acc; a <- wrap(f)) yield r ++= a
+        (acc, f) => acc.flatMap(acc => f.map(acc += _).recover{ case NonFatal(_) => acc })
```

Comment:
btw, all exceptions in `Future` are `NonFatal`, others will be exposed outside immediately

Reply:
Ah, thx!